usdt-impl/src/no-linker.rs

```diff
@@ -100,12 +110,13 @@ fn extract_probe_records_from_section() -> Result<Option<Section>, crate::Error>
         static dtrace_probes_stop: usize;
     }
 
-    // Without this the illumos linker may decide to omit symbols referencing this section.
-    // The macos linker doesn't seem to require this.
+    // Without this the illumos linker may decide to omit the symbols above that
+    // denote the start and stop addresses for this section. The macos linker
+    // doesn't seem to require this.
     #[cfg(target_os = "illumos")]
     #[link_section = "set_dtrace_probes"]
     #[used]
-    static FORCE_LOAD: [u8; 0] = [];
+    static FORCE_LOAD: [u64; 0] = [];
```

Comment:
Is the change to `u64` required? I'm curious if the linker behaves differently in either case.

Reply:
This is not necessary. I made this change because it effects sh_addralign and I thought it would be better if both the records and the placeholders had the same value there.